### ISSUE TITLE
[HELP!] IOS app crashing, If i am getting incoming call on killed state (working fine on foreground and background)

### ISSUE THREAD

**Issue by himanshupkarak** [twilio/twilio-voice-react-native#400]
# Issue
IOS app crashing, If i am getting incoming call on killed state (working fine on foreground and background)

## Pre-submission Checklist
- [x] I have verified that the issue occurs with the latest release and is not marked as a known issue in the [CHANGELOG.md](https://github.com/twilio/twilio-voice-react-native/blob/main/CHANGELOG.md).
- [x] I reviewed the [Common Issues](https://github.com/twilio/twilio-voice-react-native/blob/main/COMMON_ISSUES.md) and open GitHub issues and verified that this report represents a potentially new issue.
- [x] I am not sharing any [Personally Identifiable Information (PII)](https://www.twilio.com/docs/glossary/what-is-personally-identifiable-information-pii)
       or sensitive account information (API keys, credentials, etc.) when reporting this issue.

## Description 
*Foreground and background incoming call working fine for me but my application is getting crashed.
*I have also applied below code in App.js
React.useEffect(() => {
    const asyncFetch = async () => {
      const initializeResult = await settlePromise(voice.initializePushRegistry());
      console.log(initializeResult, 'init result push registry')
    }
    if (Platform.OS === 'ios') {
      asyncFetch();
    }
  }, [voice])

*

## Reproduction Steps
1. Make a call from any platform to IOS application logged in user's number
2. If app is in killed state it shows a popup error message like...
["your app name"] Crashed
"Do you want to share additional information with the developer?"

## Expected Behavior
it should show the notification of incoming call. same as foreground and background states

## Actual Behavior
not showing any incoming call notifications on killed state.

## Reproduction Frequency
In my case, Every time. 100%

## Screenshots
there was just a native crash dialog on IOS.

## Software and Device Information
**Please complete the following information.**
**Testing device on whice i am testing**
 - Device: [iphone 14]
 - OS:IOS
 - React version: [18.2.0]
 - React Native version: [0.72.10]
 - Node version: [18.20.3]
 - npm or yarn version: [10.7.0]

## Additional Context
no

**Comment by mhuynh5757** [comment]
Hey @himanshupkarak thanks for reaching out. Just to confirm, you're using a real, physical iOS device? You are not using the iOS simulator?

**Comment by himanshupkarak** [comment]
yes i am using physical IOS device.

**Comment by mhuynh5757** [comment]
Thanks for the additional info. Could you share more logs from Xcode with us? Especially including a stack trace, redacted of any personal information that you cannot share?

**Comment by himanshupkarak** [comment]
there is no logs in xcode while app is in killed state

**Comment by bobiechen-twilio** [comment]
Hi @himanshupkarak 

Please follow [the instructions in this discussion](https://stackoverflow.com/questions/59896343/debugging-killed-app-in-xcode-debuggernotification-debugging) to attach the Xcode debugger while launching the app from the terminated state.

**Comment by himanshupkarak** [comment]
ok, i edit Scheme in Xcode as per above documentation, but still on killed state not getting any logs in xcode.  if receive any incoming call

**Comment by himanshupkarak** [comment]
also i am getting this error in appdelegte.mm file

Use of undeclared identifier 'TwilioVoice'

**Comment by himanshupkarak** [comment]
hey @bobiechen-twilio i solved this killed state issue, but now can you tell me how to handle accept call  if accepted from killed state.

below code working for killed state

  useEffect(() => {
        voice.addListener(Voice.Event.CallInvite, handlePendingCallInvite);

        return () => {
            voice.removeListener(Voice.Event.CallInvite, handlePendingCallInvite);
        };
    }, []);
    
    
  const handlePendingCallInvite = async (callInvite) => {
        console.log(callInvite, 'invite new')
        setCallInvite(callInvite);
        const callInviteDetails = getIncomingCallInviteInfo(callInvite);
        callInvite.on(CallInvite.Event.Accepted, async (res) => {
            dispatch(setCurrentCallInfo(callInviteDetails))
            dispatch(setCallInstance(res));
            dispatch(setIncomingOrOutgoing('incoming'));
            navigation.navigate('ActiveCall', { ActiveCallDetails: res });
        });
        if (Platform.OS === 'android') {
            callInvite.on(CallInvite.Event.NotificationTapped, (call) => {
                console.log('tapped', call)
                const callInviteDetails = getIncomingCallInviteInfo(callInvite);
                dispatch(setCurrentCallInfo(callInviteDetails))
                navigation.navigate('IncomingCall', { ActiveCallDetails: callInvite })
            });
        }
    };

**Comment by mhuynh5757** [comment]
Hi @himanshupkarak I am glad to hear that you have resolved your issue. Could you please share with us your solution? Also, for you most recent message, please format your code so we are able to debug your issue.

**Comment by himanshupkarak** [comment]
Solution: I just update this piece of code in appdelegate.mm file
- (void)pushRegistry:(PKPushRegistry *)registry didReceiveIncomingPushWithPayload:(PKPushPayload *)payload forType:(PKPushType)type withCompletionHandler:(void (^)(void))completion {
   if ([type isEqualToString:PKPushTypeVoIP]) {
        NSLog(@"Received VoIP push payload: %@", payload.dictionaryPayload);
     dispatch_queue_t customQueue = dispatch_queue_create("com.dialtalk", NULL);
     [TwilioVoiceSDK handleNotification:payload.dictionaryPayload delegate:self delegateQueue:customQueue];
        
     [[NSNotificationCenter defaultCenter] postNotificationName:@"TwilioVoicePushRegistryNotification"
                                                             object:nil
                                                           userInfo:@{@"type": @"incomingPushReceived",
                                                                      @"incomingPushPayload": payload.dictionaryPayload}];
    }
    completion();
}

formated code:

useEffect(() => {
  voice.addListener(Voice.Event.CallInvite, handlePendingCallInvite);
  return () => {
    voice.removeListener(Voice.Event.CallInvite, handlePendingCallInvite);
  };
}, []);

const handlePendingCallInvite = async (callInvite) => {
  console.log(callInvite, 'invite new')
  setCallInvite(callInvite);
  const callInviteDetails = getIncomingCallInviteInfo(callInvite);
  callInvite.on(CallInvite.Event.Accepted, async (res) => {
    dispatch(setCurrentCallInfo(callInviteDetails))
    dispatch(setCallInstance(res));
    dispatch(setIncomingOrOutgoing('incoming'));
    navigation.navigate('ActiveCall', { ActiveCallDetails: res });
  });
  if (Platform.OS === 'android') {
    callInvite.on(CallInvite.Event.NotificationTapped, (call) => {
      console.log('tapped', call)
      const callInviteDetails = getIncomingCallInviteInfo(callInvite);
      dispatch(setCurrentCallInfo(callInviteDetails))
      navigation.navigate('IncomingCall', { ActiveCallDetails: callInvite })
    });
  }
};


@mhuynh5757 please help in accepting this call now   as i am trying to accept from killed state?

**Comment by mhuynh5757** [comment]
@himanshupkarak your code is still not formatted properly. Please use proper Github Markdown formatting for code blocks, using triple-backticks. https://docs.github.com/github/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax

**Comment by mhuynh5757** [comment]
Hi @himanshupkarak is issue #405 the same issue as this one? If so, we're still waiting on Xcode logs from your end for further debugging.

**Comment by ayazalphasquad** [comment]
hi @mhuynh5757 - I'm facing a similar issue with my app in the killed state. I've integrated the react-native-callkeep package along with PushKit and CallKit to handle the UI, and it works fine in the foreground and background modes. But, when the app is killed, it crashes because the listeners are in the JavaScript layer.

I am unable to view the Xcode console logs since the app disconnects with the xCode as soon as it's terminated/killed. Could you please assist me with this issue?

**Comment by mhuynh5757** [comment]
Hi @ayazalphasquad please consider opening a new issue for your problem if you are still in need of help. I'm closing this issue as stale for now.
